plugins/package-managers/go/src/main/kotlin/utils/GoSupport.kt

```diff
@@ -25,3 +25,24 @@ package org.ossreviewtoolkit.plugins.packagemanagers.go.utils
  * https://go.dev/ref/mod#incompatible-versions.
  */
 fun normalizeModuleVersion(moduleVersion: String): String = moduleVersion.removePrefix("v").substringBefore("+")
+
+/**
+ * Module paths appear as substrings of file system paths in the module cache on the file system.
+ * Go does not rely on the file system to be case-sensitive. For this reason, Go has decided to
+ * replace every uppercase letter in file system paths with an exclamation mark followed by the
+ * letter's lowercase equivalent.
+ */
+fun escapeVersion(version: String): String {
+    require(!version.contains("!")) { "Module versions must not contain exclamation marks: $version"}
+
+    val escaped = StringBuilder()
```

Comment:
please use instead of `StringBuilder` this:

```
buildString {
    append(..)
}
```

nit: maybe a to more compact could be to use the `map` function instead, up to you to decide.

Reply:
Done.